### DISCUSSION TITLE
GUI for PostgreSQL Database with NiceGUI?

### DISCUSSION THREAD

**Discussion by ghost** [zauberzeug/nicegui#392] [Q&A] [answered]
Hello,
I have a PostgreSQL database for meetings and conferences with around 20 tables and plenty of relations between the tables (people, events, rooms, etc.). So far I have made a GUI with wxPython. Now I want to switch to a WEB GUI. Can I do this with NiceGUI?
I have already found normal widgets incl. grids, that looks quit good. But what I want to realize is for example:

- Show and manipulate data in both tables and grids.
- Subforms. I.e. show the data for selection in a table on the left, manipulate the data in a form on the right.
- Subforms for 1-n relationships. So on the left a data record and on the right the n associated data records.
- Are there also comboboxes?
- How to pack the widgets? From wxPython I know a kind of hirarchical panels, where the widgets belong to, and different kinds of geometry managers for positioning all the widgets in a window/frame.

With thanks and greetings
Ulrich

**Comment by rodja** [comment, marked as answer]
> Now I want to switch to a WEB GUI. Can I do this with NiceGUI?

Yes. NiceGUI is a full featured web-UI framework.

You can structure your UI elements by using `ui.column()`, `ui.row()` or `ui.card()`. These can also be put into methods or functions for better modularity. See https://github.com/zauberzeug/nicegui/tree/main/examples/modularization for an example. Forms can be implemented with `ui.input()` and `ui.number` elements as well as toggles, combo-boxes and the like (see https://nicegui.io/reference). Use `ui.button` to trigger actions like saving the values from the elements to database.
